ASM/c/item_draw_functions.c

```diff
@@ -719,28 +721,35 @@ void draw_gi_magic_meter(z64_game_t* game, uint32_t draw_id) {
     z64_gfx_t *gfx = game->common.gfx;
 
     // Magic
-    colorRGBA8_t prim_color = item_draw_table[draw_id].args[3].color;
+    colorRGBA8_t prim_color = item_draw_table[draw_id].args[5].color;
     if (CFG_CORRECT_MODEL_COLORS) {
-        prim_color.r = CFG_MAGIC_COLOR .r;
-        prim_color.g = CFG_MAGIC_COLOR .g;
-        prim_color.b = CFG_MAGIC_COLOR .b;
+        prim_color.r = CFG_MAGIC_COLOR.r;
+        prim_color.g = CFG_MAGIC_COLOR.g;
+        prim_color.b = CFG_MAGIC_COLOR.b;
+    }
+    colorRGBA8_t env_color = item_draw_table[draw_id].args[6].color;
```

Comment:
I'm still not sure how this is working "right" but testing indicates it looks as intended, so w/e.